BO: Fix texture field not displaying for new color attribute groups

younes-bkb opened the pull request:
| Questions | Answers |
| :--- | :--- |
| **Branch?** | `9.0.x` |
| **Description?** | This PR fixes a bug in the Back Office where the "Texture" file upload field was not displayed when creating an attribute value for a newly created group of type "Color or Texture". <br><br> The root cause is a client-side JavaScript issue that incorrectly hides the form fields. This fix addresses the problem on the server-side by conditionally adding the `color` and `texture` fields in the `AttributeType.php` form builder **only** when the parent attribute group's type is `color`. This bypasses the faulty JavaScript and ensures the expected behavior. |
| **Type?** | `bug fix` |
| **Category?** | `BO` (Back Office) |
| **BC breaks?** | `no` |
| **Deprecations?** | `no` |
| **How to test?** | 1. Go to **Sell > Catalog > Attributes & Features**. <br> 2. Create a new **Attribute Group**. <br> 3. Name it (e.g., "Custom Logos") and select the type **"Color or Texture"**. Save it. <br> 4. Enter the newly created group and click **"Add new value"**. <br> 5. **Verify the bug:** Before this PR, the "Color" and "Texture" fields are missing. <br> 6. **Apply this PR.** <br> 7. Repeat step 4. <br> 8. **Verify the fix:** The form now correctly displays the "Color" and "Texture" fields. |
| **UI Tests** | |
| **Fixed issue or discussion?** | None. This PR includes the full description and steps to reproduce the bug. |
| **Related PRs** | None. |
| **Sponsor company** |  |

younes-bkb committed:
feat: implement dynamic color group detection

- Add data-color-groups attribute to pass color group IDs to frontend
- Replace hardcoded '== 2' condition with dynamic colorGroupIds.includes() check
- Solution is now future-proof for any color attribute group ID

Addresses reviewer feedback from @Hlavtox

ps-jarvis commented:
<!-- PR_TABLE_DESCRIPTION_ERROR -->
Hi, thanks for this contribution!

I found some issues with the Pull Request description:

* The `branch` should be `develop`, `9.0.x` or `8.2.x`. ([Read explanation](https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#branch))
* The `description` shouldn&#039;t be empty. ([Read explanation](https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#description))
* The `type` should be one of these: `new feature`, `improvement`, `bug fix` or `refacto`. ([Read explanation](https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#branch))
* The `category` should be one of these: `FO`, `BO`, `CO`, `IN`, `WS`, `TE`, `LO`, `ME` or `PM`. ([Read explanation](https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#branch))
* The `BC breaks` should be `yes` or `no`. ([Read explanation](https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#bc-breaks))
* The `deprecations` should be `yes` or `no`. ([Read explanation](https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#deprecations))
* The `How to test` shouldn&#039;t be empty. ([Read explanation](https://devdocs.prestashop-project.org/9/contribute/contribution-guidelines/pull-requests/#how-to-test))

Would you mind having a look at it? This will help us understand how interesting your contribution is, thank you very much!

(Note: this is an automated message, but answering it will reach a real human)

ps-jarvis commented:
<!-- PR_WELCOME -->
Hello @younes-bkb!

This is your first pull request on [PrestaShop](https://github.com/PrestaShop/PrestaShop/) repository of the PrestaShop project.

Thank you, and welcome to this Open Source community!

Hlavtox reviewed (changes requested):
Hi, I need some answers here. :-))

- I saw this issue somewhere, but it works perfectly for me on clean 9.0.0. Please link a related issue.
- You completely remove the inputs from the form, does the switching from color to non-color attribute group still work?

Hlavtox commented:
@kpodemski 
<img width="1903" height="908" alt="Snímek obrazovky 2025-08-22 213515" src="https://github.com/user-attachments/assets/70cfcfcc-7c42-42bc-86ac-5f3beb7c7de4" />

younes-bkb commented:
Hello @Hlavtox  and @kpodemski 

Thank you for your time and for looking into this.

- I have changed the base branch to `9.0.x` as requested.

- Regarding your feedback, @Hlavtox: Thank you for testing and providing the screenshot. I noticed that you renamed the default "Color" group (ID #2). This specific group works correctly, which is actually part of the issue.

The bug I am fixing occurs **only on newly created attribute groups** that are given the type "Color or Texture". The default group (ID #2) seems to be handled differently by the client-side scripts, which is why it works fine.

Could you please try to reproduce the bug by following the "How to test" steps exactly, which involves **creating a brand new group from scratch**? This is where the issue appears.

This also explains my technical choice:
When switching the attribute group in the dropdown, the form reloads the relevant fields via an AJAX call. Therefore, my initial solution (completely removing the `color` and `texture` fields from the DOM for non-color groups) **does not break the dynamic switching functionality**, but it correctly fixes the display bug for newly created groups.

Let me know if you need further information.

Hlavtox commented:
@younes-bkb I checked the code right now. I am 99% sure your solution is not the correct one. All info here - https://github.com/PrestaShop/PrestaShop/issues/38766#issuecomment-3215420305

This must be solved by providing information to the client side - which attribute is a color or not.
Then, on the JS side, replace the hardcoded `== 2` condition by `== isColor`. Then it will work.

younes-bkb commented:
@Hlavtox You're absolutely right! I've implemented exactly the solution you suggested. 

I've now:
1. Added a `data-color-groups` attribute to the attribute group select field containing all color group IDs (server-side)  
2. Replaced the hardcoded `== 2` condition with a dynamic check using `colorGroupIds.includes(value)` (client-side)

This approach is much more robust and will work for any color attribute group regardless of its ID. The fix is now fully dynamic and future-proof.

Could you please review the updated implementation?

kpodemski commented:
Hello @younes-bkb 

Thank you for contributing. The rebase you performed wasn't successful. There are multiple commits that should not be included in this pull request. Could you take a look?

younes-bkb commented:
hello @kpodemski,
Yes, sorry, this is my first open source contribution. I was on the develop branch instead of 9.0.X.

Hlavtox commented on src/PrestaShopBundle/Form/Admin/Sell/Catalog/AttributeType.php:
```diff
@@ -106,22 +110,20 @@ public function buildForm(FormBuilderInterface $builder, array $options)
-                'required' => false,
             ]);
 
+        if ($hasAttributeGroupId && $attributeGroup !== null && $attributeGroup->group_type === 'color') {
```
Should this condition be there?

Hlavtox commented:
@younes-bkb Hi, rebase looks good! :-) Now let's get back to work. I think that maybe you forgot the part of the original solution in your PR?

ChillCode commented:
Hi @younes-bkb 

I just searched the PrestaShop code to find an example using choice_attr, and there are plenty of places using it, but take a look at `/src/Core/Form/ChoiceProvider/CountryByIdChoiceProvider.php`, is the one more like my example, and check Symfony docs to see that the callback can be replaced by an array with the exact same order and values as the groups have, I just didn't want to modify the current code so added the callback.

How Country selector looks when we want to add an address:

<img width="498" height="265" alt="Captura de pantalla 2025-08-29 083655" src="https://github.com/user-attachments/assets/8241bab3-4341-4890-8315-b1fb91b2633a" />

Thanks for the PR

ChillCode commented:
Hi @younes-bkb 

Yesterday I created an alternative solution, you can check it and make it yours if you want:

https://github.com/PrestaShop/PrestaShop/compare/9.0.x...ChillCode:PrestaShop:fix/allow-attribute-group-texture

Just separated and made use of the correct repository and created a choice provider, I guess is the correct way to accomplish this, at least it works in all scenarios, when you are inside Group are when you're not inside a group, and if there are no groups created a new value cannot also be created, this part can be reworked a bit if needed inside the controller.

Hope your PR is accepted at some point, is critical.

kpodemski commented:
Hello

I think this PR can be closed because an alternative solution has been implemented, thanks to @ChillCode [here](https://github.com/PrestaShop/PrestaShop/pull/39758)

Thank you for your contribution @younes-bkb ❤️
